### PR TITLE
[@types/puppeteer] Add evaluateHandle, focus, hover, tap, type to Frame definition

### PR DESCRIPTION
**Pull request by armoniaechaos**
Please fill in this template.

Closes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/25506

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#class-frame
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/puppeteer/index.d.ts [hunk 1]
```diff
@@ -804,6 +804,15 @@ export interface FrameBase {
...
   addStyleTag(options: StyleTagOptions): Promise<void>;
 
+  /**
+   * This method fetches an element with selector, scrolls it into view if needed, and
+   * then uses `page.mouse` to click in the center of the element. If there's no element
+   * matching selector, the method throws an error.
+   * @param selector A selector to search for element to click. If there are multiple elements satisfying the selector, the first will be clicked.
+   * @param options Specifies the click options.
+   */
+  click(selector: string, options?: ClickOptions): Promise<void>;
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: Now that both `Frame` and `Page` have a `click` method we can finally remove the one for the `Page` class https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25508/files#diff-26c1cae70a0e01ff10cdc4e82232b3eaL952

**Review by marvinhagemeister — CHANGES_REQUESTED** [review]
Awesome, thank you so much for making a PR. Would add the `click` method from `page` and than it's ready to go in :) Thanks again for the PR, you rock!

File: types/puppeteer/index.d.ts (hunk 1 above)
Comment (armoniaechaos) [reply types/puppeteer/index.d.ts]: Done 👍

**Review by marvinhagemeister — APPROVED** [review]
Awesome, thanks again for the PR. This is ready to be merged 🎉

**Pull request merged**